view/SSViewer.php

```diff
@@ -819,14 +830,23 @@ public static function add_themes($themes = []) {
 	}
 
 	public static function get_themes() {
-		$res = ['$default'];
+		$default = [self::DEFAULT_THEME];
+
+		if (!Config::inst()->get('SSViewer', 'theme_enabled')) {
+			return $default;
+		}
+
+		// Explicit list is assigned
+		if ($list = Config::inst()->get('SSViewer', 'themes')) {
+			return $list;
+		}
 
-		if (Config::inst()->get('SSViewer', 'theme_enabled')) {
-			if ($list = Config::inst()->get('SSViewer', 'themes')) $res = $list;
-			elseif ($theme = Config::inst()->get('SSViewer', 'theme')) $res = [$theme, '$default'];
+		// Support legacy behaviour
+		if ($theme = Config::inst()->get('SSViewer', 'theme')) {
+			return [$theme, self::DEFAULT_THEME];
```

Comment:
In general I prefer "set a variable that returns once at the end of a function" over "return from the middle of a function" - it can be hard to see a return in the middle of a function. In this case the function is simple enough it doesn't make too much difference though.